starfish/test/image/test_slicedimage_dtype.py

```diff
@@ -63,38 +72,56 @@ def test_int_type_promotion(synthetic_stack_factory):
         )
         assert len(w) == 1
         assert issubclass(w[0].category, DataFormatWarning)
-    expected = numpy.empty(
+    expected = numpy.ones(
         (NUM_HYB,
          NUM_CH,
          NUM_Z,
          HEIGHT,
          WIDTH), dtype=numpy.int32)
-    expected.fill(16777216)
-    corner = numpy.ones(
+    corner = numpy.empty(
         (HEIGHT,
          WIDTH), dtype=numpy.int32)
+    corner.fill(16777216)
     expected[0, 0, 0] = corner
-    assert stack.numpy_array.all() == expected.all()
+    assert numpy.array_equal(stack.numpy_array, expected)
```

Comment:
if you wanted to use `.all()` here I think it would work to use it as follow, although I think `array_equal` is more expressive: 

`(stack.numpy_array == expected).all()`
or
`np.all(stack.numpy_array == expected)`